src/mlir/ast-utils.cc

```diff
@@ -34,6 +34,14 @@ namespace mlir::verona::ASTInterface
     return std::find(kinds.begin(), kinds.end(), getKind(ast)) != kinds.end();
   }
 
+  bool hasA(::ast::WeakAst ast, NodeKind kind)
+  {
+    auto nodes = ast.lock()->nodes;
+    return std::find_if(nodes.begin(), nodes.end(), [&](::ast::Ast& node) {
+             return getKind(node) == kind && !node->nodes.empty();
```

Comment:
Is there a reason that this can’t be implemented in terms of isa? 

What is the second check in this test for? Can you add a comment explaining it.

Reply:
No reason, changing. Looking at it again, the second check is more about the `else` node than the `hasA` functionality, so I'll move it down to the right place.